scripts/gee

```diff
@@ -563,17 +563,38 @@ function _startup_checks() {
   # Ensure that (most) gee commands are run from within the gee repository,
   # otherwise strange things might happen (if the user's home directory is a
   # git repository, for instance).
+
+  # 1. Ensure that a gee repository exists.
+  if [[ ! -d "${GEE_DIR}/${REPO}" ]]; then
+    _info "Directory ${GEE_DIR}/${REPO} not found, run \"gee init\" to create."
+    exit 1
+  fi
+  _set_main
+  if [[ ! -d "${GEE_DIR}/${REPO}/${MAIN}" ]]; then
+    _die "Directory ${GEE_DIR}/${REPO}/${MAIN} not found." \
+      "Somehow your ${MAIN} branch got removed?"
+    exit 1
+  fi
+
+  # 1.5. Make sure we're not in one of bazel's weird symlink directories.
+  while [[ "$(pwd -P)" != "$(pwd -L)"  ]]; do cd ..; done
+
+  # 2. Ensure that we are in a gee-controlled git repo.  If not,
+  #    switch to the main branch of one.
   local GITDIR
-  GITDIR="$(readlink -f "$("${GIT}" rev-parse --git-common-dir)")"
-  if [[ "${GITDIR}" != "${GEE_DIR}"/*/.git ]]; then
-    _warn "Error: refusing to run this gee command outside of the gee" \
-          "repository."
-    if [[ -d "${GEE_DIR}" ]]; then
-      _info "Use \"gcd\" to switch to a gee branch and try again."
-    else
-      _info "No gee repository found, run \"gee init\" to create one."
+  if ! GITDIR="$("${GIT}" rev-parse --git-common-dir 2>/dev/null )"; then
+    # not in a git directory
+    _set_main
+    _info "Switching to ${GEE_DIR}/${REPO}/${MAIN}"
```

Comment:
I'm not sold on the idea of switching from some directory outside of git into an arbitrary branch/directory just so the git command can be run.  It's ok to to fail.  Many git commands don't make sense to run outside of a git repo.  Some commands are harder to undo than others.

Reply:
I am not sure either what problem we are trying to solve here. If the user runs gee outside of the repository, what should be the expected behavior?

Reply:
expected behavior should be: if gee can do something reasonable, it should.

gee is a wrapper around git and gh.  Some git and gh commands work outside of a git repository, others don't.  Rather than deal with all of the edge cases, gee previously would refuse to run if executed from outside a gee-controlled git repository.  But that meant that innocuous commands like `gee lspr` would fail unless the user `gcd master` first.

This PR saves the user a step -- if a command is run outside of a gee-controlled git repo, we normalize the environment by switching to the master branch of the gee-controlled repo, and then proceeding.  By eliminating this variable, our use of git and gh becomes more reliable.

Reply:
You make a great point about "innocuous" commands.  What if you limit the behavior of switching into a git/gee directory to only innocuous commands?  Or only commands where gee "can do something reasonable?"

Reply:
I guess I'm arguing that there aren't any gee commands for which this is harmful?  the only one I can think of is "gee commit", which will fail with a clear error message if executed from the master branch.

So: they're all innocuous.

Reply:
Or perhaps put differently: _startup_checks is meant to run before (most) gee commands.  It's goal is to either fail with a clear error message if something in the environment isn't correct, or to "fail functional" (correct anything that needs correcting).

Reply:
Oh, that's cool.  I didn't realize.  You have specific checks that prohibit commits to master?  That's a great feature!  

And presumably this works regardless of what a team decides to call their "MAIN" branch?  What about if we have multiple special branches such as if we moved to a branch based development model?

Commit is the easiest "harmful" command to come up with but it's hard to exhaustive.  What about `reset`?  In general I would think of "read-only" commands as the only innocuous ones.  For me it makes sense to require me to be IN the repo for anything that would modify the state of my repo.

Reply:
gee does query the upstream repo metadata to get the correct name for the master branch, and falls back on "main" if all else fails.

so: even after all of my attempts to wriggle away from it, I ultimately decided that you're right, having a list of safe commands is the best option -- that way, if I add a future potentially unsafe command I'll have to think clearly about whether or not to add it to the "safe" list.  while testing, I also noticed a minor bug that was creating spurious warning messages from bash, and fixed that too.

Reply:
You forgot:
```
40 profit!!!! 🤣
```